Fix pydocstyle errors listed in issue 112589

zabboud opened the pull request:
Fixes #112589

Fixed errors relating to pydocstyle in the following files. The remaining errors are related to docstrings at the module level and at methods within each module (see details below)

pydocstyle torch/cuda/_utils.py --count
before: 3
after: 0

pydocstyle torch/cuda/jiterator.py --count
before: 3
after: 1

**remaining errors:**
```
torch/cuda/jiterator.py:1 at module level: 
        D100: Missing docstring in public module
```

pydocstyle torch/cuda/graphs.py --count   
before: 25
after: 7

**remaining errors:**
```
torch/cuda/graphs.py:1 at module level:
        D100: Missing docstring in public module
torch/cuda/graphs.py:54 in public method `__new__`:
        D102: Missing docstring in public method
torch/cuda/graphs.py:108 in public method `debug_dump`:
        D205: 1 blank line required between summary line and description (found 0)
torch/cuda/graphs.py:108 in public method `debug_dump`:
        D400: First line should end with a period (not ':')
torch/cuda/graphs.py:150 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/graphs.py:172 in public method `__enter__`:
        D105: Missing docstring in magic method
torch/cuda/graphs.py:186 in public method `__exit__`:
        D105: Missing docstring in magic method
```

pydocstyle torch/cuda/_sanitizer.py --count   
before: 35
after: 31

**remaining errors:**
```
torch/cuda/_sanitizer.py:43 in public class `AccessType`:
        D101: Missing docstring in public class
torch/cuda/_sanitizer.py:47 in public method `__str__`:
        D105: Missing docstring in magic method
torch/cuda/_sanitizer.py:84 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:96 in public method `__str__`:
        D105: Missing docstring in magic method
torch/cuda/_sanitizer.py:139 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:142 in public method `__str__`:
        D105: Missing docstring in magic method
torch/cuda/_sanitizer.py:218 in public class `StreamSynchronizations`:
        D101: Missing docstring in public class
torch/cuda/_sanitizer.py:219 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:256 in public method `create_stream`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:268 in public method `create_event`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:272 in public method `delete_event`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:276 in public method `update_seq_num`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:280 in public method `record_state`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:291 in public method `stream_wait_for_event`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:298 in public method `all_streams_wait_for_event`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:307 in public method `all_streams_wait_for_stream`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:316 in public method `sync_all_streams`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:323 in public method `is_ordered_after`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:339 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:460 in public function `zip_by_key`:
        D103: Missing docstring in public function
torch/cuda/_sanitizer.py:466 in public function `zip_arguments`:
        D103: Missing docstring in public function
torch/cuda/_sanitizer.py:478 in public class `ArgumentHandler`:
        D101: Missing docstring in public class
torch/cuda/_sanitizer.py:479 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:505 in public method `parse_inputs`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:520 in public method `parse_outputs`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:527 in public class `CUDASanitizerDispatchMode`:
        D101: Missing docstring in public class
torch/cuda/_sanitizer.py:528 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:562 in public method `__torch_dispatch__`:
        D105: Missing docstring in magic method
torch/cuda/_sanitizer.py:597 in public method `__init__`:
        D107: Missing docstring in __init__
torch/cuda/_sanitizer.py:601 in public method `enable`:
        D102: Missing docstring in public method
torch/cuda/_sanitizer.py:605 in public method `__del__`:
        D105: Missing docstring in magic method
```

pydocstyle torch/storage.py --count   
before: 90
after: 37

**remaining errors:**
```
torch/storage.py:1 at module level:
        D100: Missing docstring in public module
torch/storage.py:310 in public class `UntypedStorage`:
        D101: Missing docstring in public class
torch/storage.py:311 in public method `__getitem__`:
        D105: Missing docstring in magic method
torch/storage.py:317 in public method `is_cuda`:
        D102: Missing docstring in public method
torch/storage.py:321 in public method `is_hpu`:
        D102: Missing docstring in public method
torch/storage.py:325 in public method `share_memory_`:
        D102: Missing docstring in public method
torch/storage.py:444 in public class `TypedStorage`:
        D101: Missing docstring in public class
torch/storage.py:453 in public method `fill_`:
        D102: Missing docstring in public method
torch/storage.py:458 in public method `__new__`:
        D102: Missing docstring in public method
torch/storage.py:530 in public method `__init__`:
        D107: Missing docstring in __init__
torch/storage.py:599 in public method `is_cuda`:
        D102: Missing docstring in public method
torch/storage.py:604 in public method `is_hpu`:
        D102: Missing docstring in public method
torch/storage.py:624 in public method `__len__`:
        D105: Missing docstring in magic method
torch/storage.py:653 in public method `__setitem__`:
        D105: Missing docstring in magic method
torch/storage.py:681 in public method `__getitem__`:
        D105: Missing docstring in magic method
torch/storage.py:715 in public method `copy_`:
        D102: Missing docstring in public method
torch/storage.py:723 in public method `nbytes`:
        D102: Missing docstring in public method
torch/storage.py:731 in public method `type`:
        D102: Missing docstring in public method
torch/storage.py:744 in public method `cuda`:
        D102: Missing docstring in public method
torch/storage.py:751 in public method `hpu`:
        D102: Missing docstring in public method
torch/storage.py:758 in public method `element_size`:
        D102: Missing docstring in public method
torch/storage.py:766 in public method `get_device`:
        D102: Missing docstring in public method
torch/storage.py:770 in public method `__str__`:
        D105: Missing docstring in magic method
torch/storage.py:781 in public method `__repr__`:
        D105: Missing docstring in magic method
torch/storage.py:785 in public method `__iter__`:
        D105: Missing docstring in magic method
torch/storage.py:789 in public method `__copy__`:
        D105: Missing docstring in magic method
torch/storage.py:793 in public method `__deepcopy__`:
        D105: Missing docstring in magic method
torch/storage.py:801 in public method `__sizeof__`:
        D105: Missing docstring in magic method
torch/storage.py:877 in public method `device`:
        D102: Missing docstring in public method
torch/storage.py:881 in public method `size`:
        D102: Missing docstring in public method
torch/storage.py:891 in public method `pickle_storage_type`:
        D102: Missing docstring in public method
torch/storage.py:902 in public method `__reduce__`:
        D105: Missing docstring in magic method
torch/storage.py:907 in public method `data_ptr`:
        D102: Missing docstring in public method
torch/storage.py:915 in public method `resize_`:
        D102: Missing docstring in public method
torch/storage.py:931 in public method `from_buffer`:
        D102: Missing docstring in public method
torch/storage.py:1032 in public method `from_file`:
        D402: First line should not be the function's "signature"
torch/storage.py:1075 in public method `is_shared`:
        D102: Missing docstring in public method

```

zabboud committed:
fixing 42 errors relating to pydocstyle

zabboud committed:
fix 4 errors relating to pydocstyle

zabboud committed:
resolve 18 errors relating to pydocstyle

zabboud committed:
fix all 3 errors relating to pydocstyle

zabboud committed:
resolve errors related to space after fn docstring

zabboud committed:
fix 12 errors with pydocstyle

zabboud commented:
@pytorchbot label "docathon-h2-2023"

zabboud commented:
@mikaylagawarecki @albanD @svekars @jbschlosser

kit1980 commented:
@zabboud please resolve the merge conflict.

kit1980 commented on torch/storage.py:
```diff
@@ -968,69 +968,69 @@ def _to(self, dtype):
     def from_file(cls, filename, shared, size):
         """
-        from_file(filename, shared=False, size=0) -> Storage
+        from_file(filename, shared=False, size=0) -> Storage.
```
I wonder if this change prevent the proper doc rendering - see https://pytorch.org/docs/stable/storage.html#torch.TypedStorage.from_file

We can verify after the merge conflict resolved.

zabboud replied:
I'll remove the period since the signature should be maintained

zabboud committed:
Update storage.py

Removing period from first line in the docstring to avoid proper Sphinx rendering

zabboud committed:
resolve conflict

Update docstring in share_memory_() to the latest branch

zabboud committed:
Merge branch 'main' into issue_112589

zabboud commented:
> @zabboud please resolve the merge conflict.

conflict has been resolved.

svekars commented:
There is still [a few lint errors](https://github.com/pytorch/pytorch/actions/runs/6804736893/job/18503628608?pr=113227) here but should be possible to autofix fix `lintrunner -a` or `quicklint`:

zabboud commented:
@svekars I ran `lintrunner -a` on branch issue_112589, but no lint issues were found - perhaps might need to be rebased?

mikaylagawarecki commented:
Did you run `make quicklint` before `lintrunner -a`?

zabboud commented:
> Did you run `make quicklint` before `lintrunner -a`?

@mikaylagawarecki No because what @svekars mentioned is to autofix with `lintrunner -a` OR `quicklint` (see below)

> There is still [a few lint errors](https://github.com/pytorch/pytorch/actions/runs/6804736893/job/18503628608?pr=113227) here but should be possible to autofix with `lintrunner -a` or `quicklint`:

zabboud committed:
Merge branch 'pytorch:main' into issue_112589

zabboud committed:
linting _sanitizer.py

kit1980 commented:
@zabboud please don't update the submodules.

zabboud committed:
Revert third_party commits

This reverts commit 0e33c09adefaae0d7b2b5740127188e806718cad.

zabboud commented:
> @zabboud please don't update the submodules.

@kit1980  apologies - I've reverted those changes, they came up with the automated linter.

zabboud commented:
@kit1980 @svekars all checked have passed - can the merge be initiated to close this PR?

svekars commented on torch/cuda/graphs.py:
```diff
@@ -53,8 +51,7 @@ def __new__(cls):
     def capture_begin(self, pool=None, capture_error_mode="global"):
-        r"""
-        Begins capturing CUDA work on the current stream.
+        r"""Begins capturing CUDA work on the current stream.
```
```suggestion
        r"""Begin capturing CUDA work on the current stream.
```

svekars commented on torch/cuda/graphs.py:
```diff
@@ -84,29 +81,23 @@ def capture_end(self):
-        r"""
-        Replays the CUDA work captured by this graph.
-        """
+        r"""Replays the CUDA work captured by this graph."""
```
```suggestion
        r"""Replay the CUDA work captured by this graph."""
```

svekars committed:
Update torch/cuda/graphs.py

svekars committed:
Update torch/cuda/graphs.py

kit1980 commented:
@pytorchbot merge
